Formula/k/kubetail.rb

```diff
@@ -0,0 +1,38 @@
+# typed: false
+# frozen_string_literal: true
+
+# This file was generated by GoReleaser. DO NOT EDIT.
+class Kubetail < Formula
+  desc "CLI tool with a web dashboard for viewing Kubernetes logs"
+  homepage "https://www.kubetail.com/"
+  url "https://github.com/kubetail-org/kubetail/archive/refs/tags/0.8.0-rc2.tar.gz"
+  version "0.0.1"
+  sha256 "6b6ac04a74a1724303e1c023ebd311d80ee10d13af922b4879a17bde4dc614ff"
+  license "Apache-2.0"
+
+  # Build dependencies
+  depends_on "go" => :build
+  depends_on "make" => :build
+  depends_on "node" => :build
+  depends_on "pnpm" => :build
+
+  def install
+    # Run the makefile build process if it orchestrates pnpm and go builds
+    system "make", "build"
+
+    # Install binary from `bin` folder
+    bin.install "bin/kubetail"
+  end
+
+  def post_install
+    generate_completions_from_executable(bin/"kubetail", "completion")
+  end
+
+  test do
+    # Verify that the kubetail binary is executable and returns expected output for --version
+    assert_match version.to_s, shell_output("#{bin}/kubetail --version")
+
+    # Alternatively, you can test with the --help flag
+    assert_match "Usage:", shell_output("#{bin}/kubetail --help")
```

Comment:
we need a more substantial test

We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate teopenjdk@8_1.8.0-432ath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Added a test using the `kubetail serve --test` command which starts and stops the web dashboard server.